avogadro/qtplugins/openbabel/obfileformat.cpp

```diff
@@ -112,6 +122,15 @@ bool OBFileFormat::read(std::istream& in, Core::Molecule& molecule)
   if (formats2D.contains(QString::fromStdString(m_fileExtensions.front())))
     options << "--gen3d";
 
+  // Check if we have extra arguments for open babel
+  json extraArgs = opts.value("OBExtraArgs", json::object());
```

Comment:
Why is this OB specific? Shouldn’t we have one key/value for arguments to pass to file formats, regardless of type? Even if the key name is documented, this seems un-necessarily opaque.

I’d prefer to have something like “formatArguments”

Reply:
Well, so far in our usage of the format options, each format has it's own keys, and each key represents a particular option for that format. See [here](https://github.com/OpenChemistry/avogadrolibs/blob/master/avogadro/io/cjsonformat.cpp#L376) and [here](https://github.com/OpenChemistry/avogadrolibs/blob/master/avogadro/io/xyzformat.cpp#L154) for examples. 

So, for the xyz format, the `perceiveBonds` key has a boolean value (true or false depending on whether we should perceive bonds or not).

For Open Babel, I pictured the json options looking like this:

```
{
  "OBExtraArgs": [
    "-xw",
    "-xz"
  ]
}
```

The type of the value for the `OBExtraArgs` key would be an array of strings. Each string gets forwarded directly as a command line argument to `obabel` (The example above is an example for the POSCAR format. Passing both of these arguments will change the way the POSCAR output is ordered).

I named it `OBExtraArgs` because I don't think we could use the values for that key for any other format. They are command line arguments for Open Babel.

Reply:
It might be fine to have particular keys for internal formats. But consider that there are two XYZ readers and two POSCAR readers now:
- Avogadro internal
- Open Babel

I understand the syntax for the JSON and that it’d be an array of strings. But picture it from a new script user - they want to write a command that uses OB’s FASTA support to generate DNA coordinates.
- How would they know to use OBExtraArgs
- Why is the key `OBExtraArgs` when for XYZ there’s a `perceiveBonds`

So my main point is that this should have a better name like `formatArguments`. I’m open to something better, but `OBExtraArgs` is IMHO a bit hacky.

Reply:
Why not just use a key called arguments? At the end of the day whoever is writing this needs to know what the arguments are, but there is no need to encode ob into the key name. Where they can be standardized for internal formats they should be, I think we can safely use the arguments keyword for passing arguments to command line tools. If internal formats reuse that in the future they can do some validation/graceful handling.

Reply:
Okay, sounds good! I amended the commit and changed the key to "arguments".